flake.nix

```diff
@@ -18,48 +18,61 @@
 
       waylandOverlay = (final: prev:
         let
-          githubTemplate = { attrName, extra ? {} }: prev.callPackage ./templates/github-template.nix { inherit prev attrName extra; };
-          waylandPkgs = rec {
+          githubTemplate = { attrName, extra ? { } }: prev.callPackage ./templates/github-template.nix { inherit prev attrName extra; };
+          checkMutuallyExclusive = lib.mutuallyExclusive (map (e: e.attrName) attrsExtraChangesNeeded) (map (e: e.attrName) attrsNoExtraChangesNeeded);
+          genPackagesGH =
+            if checkMutuallyExclusive then
+              lib.listToAttrs
+                (map
+                  (a: {
+                    name = a.attrName;
+                    value = githubTemplate a;
```

Comment:
TODO: think of a way to have gitlab and sourcehut templates.

maybe just have the nix equivalent of a match block in template.nix which uses the domain field in metadata.nix